### PR TITLE
feat(tests): add test case for create2 selfdestruct and recreate

### PR DESCRIPTION
**Pull request by nerolation**
🗒️ Description

This PR adds a BAL test for a EIP-6780 edge case: self-destructing a contract and recreating it at the same address in a later transaction within the same block.

## 🔗 Related Issues or PRs

N/A.

## ✅ Checklist

- [ ] All: Ran fast `tox` checks to avoid unnecessary CI fails, see also [Code Standards](https://eest.ethereum.org/main/getting_started/code_standards/) and [Enabling Pre-commit Checks](https://eest.ethereum.org/main/dev/precommit/):
    ```console
    uvx tox -e static
    ```
- [ ] All: PR title adheres to the [repo standard](https://eest.ethereum.org/main/getting_started/contributing/?h=contri#commit-messages-issue-and-pr-titles) - it will be used as the squash commit message and should start `type(scope):`.
- [ ] All: Considered updating the online docs in the [./docs/](/ethereum/execution-specs/blob/HEAD/docs/) directory.
- [ ] All: Set appropriate labels for the changes (only maintainers can apply labels).
- [ ] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://eest.ethereum.org/main/tests/) are correctly formatted.
- [ ] Tests: For PRs implementing a missed test case, update the [post-mortem document](/ethereum/execution-specs/blob/HEAD/docs/writing_tests/post_mortems.md) to add an entry the list.
- [ ] Ported Tests: All converted JSON/YML tests from [ethereum/tests](/ethereum/tests) or [tests/static](/ethereum/execution-specs/blob/HEAD/tests/static) have been assigned `@ported_from` marker.

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

File: tests/amsterdam/eip7928_block_level_access_lists/test_cases.md [hunk 1]
```diff
@@ -131,3 +131,5 @@
...
... The created contract optionally does `SLOAD/SSTORE` (to prove state touches), then `SELFDESTRUCT(beneficiary=B)` **in the same tx** (so under EIP-6780 the account is actually deleted after Tx1). **Tx2:** `Factory` executes the *same* `CREATE2(salt, initcode)` again, recreating **A**, and this time the runtime code persists (no SELFDESTRUCT). | BAL **MUST** include: **Tx1 (`block_access_index=1`)**: (1) `Factory` with `nonce_changes` (CREATE2 increments), and `balance_changes` if it endows A. (2) **A** in `account_changes` (it was accessed/created) but **MUST NOT** have persistent `code_changes`, `nonce_changes`, or `storage_changes` (it ends Tx1 non-existent due to same-tx create+SELFDESTRUCT). Any attempted `SSTORE` in A before SELFDESTRUCT **MUST NOT** appear in `storage_changes` (ephemeral). (3) `B` with `balance_changes` if A had balance transferred on SELFDESTRUCT. **Tx2 (`block_access_index=2`)**: (1) `Factory` with another `nonce_changes`. (2) **A** with `code_changes` (runtime bytecode present), `nonce_changes = 1`, plus any `storage_changes` performed in Tx2. (3) If Tx2 endows or transfers value, include corresponding `balance_changes` for involved accounts. | 🟡 Planned |
```
Comment (jochem-brouwer) [inline tests/amsterdam/eip7928_block_level_access_lists/test_cases.md]: > and `balance_changes` if it endows A.

(1) `balance_changes` should be B if it endows B right? (`SELFDESTRUCT(beneficiary=B)`)

Comment (jochem-brouwer) [reply tests/amsterdam/eip7928_block_level_access_lists/test_cases.md]: I think it should be parameterized for existing A or not (A has balance or not). If A has a balance before tx1, then in tx1 it gets destroyed and in tx2 it gets resurrected. If it does not have balance before tx1, then the account never existed, so from an external point of view it looks like A is created in tx2.

Comment (nerolation) [reply tests/amsterdam/eip7928_block_level_access_lists/test_cases.md]: Fixed

**Review by fselmo — APPROVED** [review]
lgtm, we can parametrize the first CREATE with `with_all_create_opcodes` to do `CREATE` and `CREATE2`, then resurrect with `CREATE2`. When this is implemented we can do another pass in case more cases arise. Great case 👍🏼
